main/java/mjc/Euro.java

```diff
@@ -0,0 +1,59 @@
+package main.java.mjc;
+
+import java.util.Locale;
+
+public class Euro implements Comparable<Euro> {
+    private final int value;
+
+    public Euro(int cents) {
+        this.value = cents;
+    }
+
+    public Euro(int euros, int cents) {
+        this.value = euros * 100 + cents;

```

Comment:
```suggestion
        this(euros * 100 + cents);
```
try using a primary constructor.